unified_planning/io/anml_writer.py

```diff
@@ -0,0 +1,330 @@
+                # For construction in the Problem, the father of a UserType is always added before the UserType itself.
+                father = cast(_UserType, t).father
+                assert father is not None
+                assert names_mapping[father] is not None
+                out.write(f' < {names_mapping[father]};\n')
+        static_fluents = self.problem.get_static_fluents()
+        for f in self.problem.fluents:
+            if f in static_fluents:
+                out.write(f'constant {_get_anml_name(f.type, names_mapping)} {_get_anml_name(f, names_mapping)};\n')
+            else:
+                out.write(f'fluent {_get_anml_name(f.type, names_mapping)} {_get_anml_name(f, names_mapping)};\n')
+
+
+        converter = ConverterToANMLString(names_mapping, self.problem.env)
+
+        for a in self.problem.actions:
+            if isinstance(a, up.model.InstantaneousAction):
+                parameters = [f'{_get_anml_name(ap.type, names_mapping)} {_get_anml_name(ap, names_mapping)}' for ap in a.parameters]
+                out.write(f'action {_get_anml_name(a, names_mapping)}({", ".join(parameters)}) {{\n')
+                for p in a.preconditions:
+                    out.write(f'   [ start ] {converter.convert(p)};\n')
+                for e in a.effects:
+                    out.write(f'   {self._convert_effect(e, converter, None, 3)}')
+                out.write('}\n')
+            elif isinstance(a, DurativeAction):
+                parameters = [f'{_get_anml_name(ap.type, names_mapping)} {_get_anml_name(ap, names_mapping)}' for ap in a.parameters]
+                out.write(f'action {_get_anml_name(a, names_mapping)}({", ".join(parameters)}) {{\n')
+                left_bound = ' > ' if a.duration.is_left_open() else ' >= '
+                right_bound = ' < ' if a.duration.is_right_open() else ' <= '
+                out.write(f'   duration{left_bound}{converter.convert(a.duration.lower)} and ')
+                out.write(f'duration{right_bound}{converter.convert(a.duration.upper)}\n')
+                for i, cl in a.conditions.items():
+                    for c in cl:
+                        out.write(f'   {self._convert_anml_interval(i)} {converter.convert(c)};\n')
+                for ti, el in a.effects.items():
+                    for e in el:
+                        out.write(f'   {self._convert_effect(e, converter, ti, 3)}')
+                out.write('}\n')
+            else:
+                raise NotImplementedError
+
+        for t in self.problem.user_types: # Define objects
+            objects: List['unified_planning.model.Object'] = list(self.problem.objects(t))
+            obj_names = [_get_anml_name(o, names_mapping) for o in self.problem.objects(t)]
+            if len(objects) > 0:
+                out.write(f'instance {_get_anml_name(t, names_mapping)} {", ".join(obj_names)};\n')
+
+        for fe, v in self.problem.initial_values.items():
```

Comment:
Just a question, are we sure to print all the initial values even when there are defaults?

Reply:
Isn't this the only option? Because using the * to define a default behaviour should be possible only in Tamer dialect. Right?

Reply:
Yes. My question was whether the `initial_values` completed the initial states if defaults are used, but I checked and it does